core/src/main/java/io/grpc/internal/PickFirstLoadBalancer.java

```diff
@@ -69,7 +70,7 @@ public void onSubchannelState(ConnectivityStateInfo stateInfo) {
 
       // The channel state does not get updated when doing name resolving today, so for the moment
       // let LB report CONNECTION and call subchannel.requestConnection() immediately.
-      helper.updateBalancingState(CONNECTING, new Picker(PickResult.withSubchannel(subchannel)));
+      updateBalancingState(CONNECTING, new Picker(PickResult.withSubchannel(subchannel)));
```

Comment:
We could include these, but it seems more bug-prone and not necessarily better behavior. I think we want to stick with subchannel updates for now. So we just need to update processSubchannelState().

(handleNameResolutionError is the only case here that really matters, and if we get addresses after having none, transitioning to CONNECTING seems fair since we don't have any evidence yet on whether the addresses work or not.)

Reply:
My instinct was to try to guarantee that `currentState` always reflected the current sate. Now only updating it in `processSubchannelState()`.

Reply:
I think your instinct was fair. It is more of `subchannelState` now, but not because we ignore updates. Dunno.

Reply:
I went ahead and reverted the last commit so that the currentStatus refers to the LB status. We don't really need to track the status changes everywhere, but the logic is more consistent this way.

Reply:
it looks the problem is the edge case whether the transient failure due to name resolution error should be sticky or not - maybe it should, it looks Round Robin does that too. Do we have a design for that? Otherwise might be fine either way we can patch if there is a real issue.

Reply:
it looks the latest behavior is that name resolution error followed by a resolution result is not sticky transient_failure.

Reply:
Sorry, merged before I saw your comments.

I believe we just care about the case when the subchannel state would transition from TF to either IDLE or CONNECTING. Name resolution errors do not have to be sticky.